main.yml

```diff
@@ -400,6 +400,44 @@ parameters:
           # Pull in the APPUiO managed info labels
           * on(cluster_id) group_left(sales_order) appuio_managed_info{
               cloud_provider=~"%(cloud_provider)s",
+              distribution=~"openshift4",
+              vshn_service_level="%(vshn_service_level)s",
+          }
+
+      appuio_managed_cores:
+        enabled: true
+        prometheus_org_id: appuio-managed-openshift-billing
+        products:
+          - product_id: 'openshiftoke-baremetal-core-besteffort'
+            params:
+              cloud_provider: baremetal
+              vshn_service_level: best_effort
+          - product_id: 'openshiftoke-baremetal-core-guaranteed'
+            params:
+              cloud_provider: baremetal
+              vshn_service_level: guaranteed_availability
+        instance_id_pattern: 'ocp-cores-%(cluster_id)s-%(role)s'
+        item_group_description_pattern: 'APPUiO Managed OpenShift - Cluster: %(cluster_id)s'
+        unit_id: 'uom_uom_198_02905845' # Core-Hour
+        query_pattern: |
+          # Calculates cores for app nodes of a cluster
+          # Max values over one hour.
+          max_over_time(
+            # Sum the vCPUs by cluster
+            sum by(cluster_id, role) (
+                # Get the node vCPUs
+                node_cpu_info
+                # Limit to worker nodes only
+                * on (cluster_id, instance) group_left(role) (
+                    # node_cpu_info and kube_node_role use different labels to identify the node.
+                    label_join(kube_node_role{role="worker"}, "instance", "", "node")
+                )
+            )[59m:1m]
+          )
+          # Pull in the APPUiO managed info labels
+          * on(cluster_id) group_left(sales_order) appuio_managed_info{
+              cloud_provider=~"%(cloud_provider)s",
+              distribution=~"oke",
```

Comment:
There's nothing that prevents us from doing full OCP4 on bare-metal, so this needs to be parameterized (either now or later on). I'd vote for now, so we don't have issues with migrating the query at a later point. We'll have to implement the same for `appuio_managed_vcpu` anyway once we start doing OKE on cloud providers.